packages/web3/src/codec/txscript-codec.ts

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2018 - 2022 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { Buffer } from 'buffer/'
+import { Parser } from 'binary-parser'
+import { DecodedArray } from './array-codec'
+import { Codec } from './codec'
+import { DecodedMethod, Method, MethodCodec, methodsCodec } from './method-codec'
+
+export interface DecodedTxScript {
+  methods: DecodedArray<DecodedMethod>
+}
+
+export interface TxScript {
+  methods: Method[]
+}
+
+export class TxScriptCodec implements Codec<DecodedTxScript> {
+  parser = Parser.start().nest('methods', {
+    type: methodsCodec.parser
+  })
+
+  encode(input: DecodedTxScript): Buffer {
+    return methodsCodec.encode(input.methods.value)
+  }
```

Comment:
Does it make sense to add an `encodeTxScript(script: TxScript)` method?

Reply:
Good idea, added now.